### ISSUE TITLE
On Windows: Doesn't work if $EDITOR has spaces

### ISSUE THREAD

**Issue by sk22** [itchyny/mmv#19]
When setting $EDITOR to `C:\Program Files\Microsoft VS Code\Code.exe`, I get this:

```
C:\Users\user\Downloads\mmv_v0.1.4_windows_amd64>mmv *
mmv: abort renames: exec: "C:\\Program": file does not exist
```

(Regarding VS Code, I should note that "C:\Progra~1\Micros~1\Code.exe" works, but Code apparently "detatches" the newly created process if there's already an instance open. Doesn't appear to happen on Linux, but I suppose this is a problem with VS Code)

**Comment by itchyny** [comment]
Hmm, this looks to work before merging #10, which allows command line flags in $EDITOR. Not sure how to fix both.

**Comment by sk22** [comment]
ahh, hm, i see... in this case, i guess `C:\Progra~1\Micros~1\Code.exe --wait` will work for me, even if it's not ideal (though very retro)

**Comment by itchyny** [comment]
Using shell could help us? You can configure editor with space path and arguments like `export EDITOR="'C:\Program Files\Microsoft VS Code\Code.exe' --wait"` (note the single quotes). This looks like how editor config is handled in git (ref: https://git-scm.com/book/en/v2/Appendix-C%3A-Git-Commands-Setup-and-Config).
```diff
diff --git a/cmd/mmv/main.go b/cmd/mmv/main.go
index 8aed5bb..9a627fe 100644
--- a/cmd/mmv/main.go
+++ b/cmd/mmv/main.go
@@ -109,10 +109,8 @@ func rename(args []string) error {
 	if editor == "" {
 		editor = "vi"
 	}
-	editorWithArgs := strings.Fields(editor)
-	editorWithArgs = append(editorWithArgs, f.Name())
 
-	cmd := exec.Command(editorWithArgs[0], editorWithArgs[1:]...)
+	cmd := exec.Command("sh", "-c", editor+` "$@"`, "", f.Name())
 	cmd.Stdin = tty.Input()
 	cmd.Stdout = tty.Output()
 	cmd.Stderr = tty.Output()
```

**Comment by itchyny** [comment]
Another patch to reduce the code by resolving the environment in the shell.
```diff
diff --git a/cmd/mmv/main.go b/cmd/mmv/main.go
index 8aed5bb..1a97724 100644
--- a/cmd/mmv/main.go
+++ b/cmd/mmv/main.go
@@ -105,14 +105,7 @@ func rename(args []string) error {
 	}
 	defer tty.Close()
 
-	editor := os.Getenv("EDITOR")
-	if editor == "" {
-		editor = "vi"
-	}
-	editorWithArgs := strings.Fields(editor)
-	editorWithArgs = append(editorWithArgs, f.Name())
-
-	cmd := exec.Command(editorWithArgs[0], editorWithArgs[1:]...)
+	cmd := exec.Command("sh", "-c", `eval exec ${EDITOR:-vi} '"$@"'`, "", f.Name())
 	cmd.Stdin = tty.Input()
 	cmd.Stdout = tty.Output()
 	cmd.Stderr = tty.Output()
```

**Comment by itchyny** [comment]
Released as [v0.1.5](https://github.com/itchyny/mmv/releases/tag/v0.1.5).

**Comment by sk22** [comment]
i suppose this works on linux, but it broke windows compatibility :/

```
C:\Users\user\Downloads\mmv_v0.1.5_windows_amd64>mmv *
mmv: abort renames: exec: "sh": executable file not found in %PATH%
```

sorry about not commenting earlier - i didn't find the time to try the new version until now

**Comment by itchyny** [comment]
I fixed in the main branch. Make sure to confirm the issue is resolved in your environment.

**Comment by sk22** [comment]
that works, thanks!

**Comment by itchyny** [comment]
Released as [v0.1.6](https://github.com/itchyny/mmv/releases/tag/v0.1.6).
